src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -1633,6 +1631,68 @@ let getSelectedBuildingBlockCell (table: Table) (context: RequestContext) =
         )
     }
 
+/// <summary>
+/// Get the main column of the arc table of the selected building block of the active annotation table
+/// </summary>
+let getArcMainColumn (excelTable: Table) (arcTable: ArcTable) (context: RequestContext) =
+    promise {
+        let! selectedBlock = getSelectedBuildingBlock excelTable context
+
+        let protoHeaders = excelTable.getHeaderRowRange()
+        let _ = protoHeaders.load(U2.Case2 (ResizeArray(["values"])))
+
+        do! context.sync().``then``(fun _ -> ())
+
+        let headers = protoHeaders.values.Item 0 |> Array.ofSeq |> Array.map (fun c -> c.ToString())
+
+        let arcTableIndices = (groupToBuildingBlocks headers) |> Array.ofSeq |> Array.map (fun i -> i |> Array.ofSeq)
+
+        let arcTableIndex =
+            let potResult = arcTableIndices |> Array.mapi (fun i c -> i, c |> Array.tryFind (fun (_, s) -> s = snd selectedBlock.[0]))
+            let result = potResult |> Array.filter (fun (_, c) -> c.IsSome) |> Array.map (fun (i, c) -> i, c.Value)
+            Array.tryHead result
+
+        let arcTableIndex, columnName =
+            if arcTableIndex.IsSome then
+                fst arcTableIndex.Value, snd (snd arcTableIndex.Value)
+            else failwith "Could not find a fitting arc table index"
+
+        let targetColumn =
+            let potColumn = arcTable.GetColumn arcTableIndex
+            if columnName.Contains(potColumn.Header.ToString()) then potColumn
+            else failwith "Could not find a fitting arc table index with matchin name"
+
+        return (targetColumn, arcTableIndex)
+    }
+
+/// <summary>
+/// Get the valid cell type for the conversion based on input cell type
+/// </summary>
+/// <param name="cellType"></param>
+let getValidConversionCellType (cellType: CompositeCellDiscriminate option) =
+    Excel.run(fun context ->
+        promise {
+            let! result = tryGetActiveAnnotationTable context
+            match result with
+            | Result.Ok excelTable ->
+                let! arcTable = ArcTable.tryGetFromExcelTable(excelTable, context)
+                let! (arcMainColumn, _) = getArcMainColumn excelTable arcTable.Value context
+
+                return
+                    match cellType with
+                    | Some CompositeCellDiscriminate.Unitized -> Some CompositeCellDiscriminate.Term
```

Comment:
This function must be reworked.

Here is what you do as far as i understand it:

goal: Return valid celltypes for conversion based on `cellType`

1. get active table
2. try parse to arctable, then take `.Value` without error handling
3. get selected main column
4. check if valid conversion

BUT;
- if `cellType` is term return unitized without any checking against the selected building block
- Similiar for `cellType` = Data

Reply:
fixed (hopefully)